src/index.ts

```diff
@@ -238,7 +239,7 @@ export const useCourseSearch = (
     return () => {
       unlisten()
     }
-  })
+  }, [])
```

Comment:
I think you'll need to either add `initSearch` here or add a comment to suppress the lint error. It should be fine to suppress it and leave the `[]` empty, since the `initSearch` function is only dependent on `setState` values which shouldn't change

Reply:
Sure, just pushed a commit for it